### ISSUE TITLE
votable: java.lang.IllegalStateException in SerializerTest with OpenJDK11

### ISSUE THREAD

**Issue by olebole** [Starlink/starjava#24]
An attempt to build the Debian votable jar package with OpenJDK 11 resulted in the following failure during test;
```
java.lang.IllegalStateException: Can't overwrite cause with java.io.FileNotFoundException: stest10126905822765568439.dat (No such file or directory)
  at java.base/java.lang.Throwable.initCause(Throwable.java:462)
  at uk.ac.starlink.votable.VOTableDOMBuilder$Worker.processFitsHref(VOTableDOMBuilder.java:156)
  at uk.ac.starlink.votable.SkeletonDOMBuilder$BasicContentHandler.startElement(SkeletonDOMBuilder.java:192)
  at uk.ac.starlink.votable.CustomDOMBuilder.startElement(CustomDOMBuilder.java:157)
  at uk.ac.starlink.votable.VOTableDOMBuilder.startElement(VOTableDOMBuilder.java:79)
  at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.startElement(AbstractSAXParser.java:510)
  at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractXMLDocumentParser.emptyElement(AbstractXMLDocumentParser.java:183)
  at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanStartElement(XMLDocumentFragmentScannerImpl.java:1377)
  at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(XMLDocumentFragmentScannerImpl.java:2708)
  at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:605)
  at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:534)
  at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:888)
  at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:824)
  at java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)
  at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1216)
  at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:635)
  at uk.ac.starlink.votable.VOElementFactory.parseToDOM(VOElementFactory.java:536)
  at uk.ac.starlink.votable.VOElementFactory.transformToDOM(VOElementFactory.java:391)
  at uk.ac.starlink.votable.VOElementFactory.makeVOElement(VOElementFactory.java:221)
  at uk.ac.starlink.votable.SerializerTest.checkBadSource(SerializerTest.java:135)
  at uk.ac.starlink.votable.SerializerTest.exerciseStreamSerializer(SerializerTest.java:118)
  at uk.ac.starlink.votable.SerializerTest.testURLs(SerializerTest.java:84)
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Caused by: org.xml.sax.SAXParseException; lineNumber: 8; columnNumber: 47; stest10126905822765568439.dat (No such file or directory)
  at uk.ac.starlink.votable.VOTableDOMBuilder$Worker.processFitsHref(VOTableDOMBuilder.java:155)
Caused by: java.io.FileNotFoundException: stest10126905822765568439.dat (No such file or directory)
  at java.base/java.io.FileInputStream.open0(Native Method)
  at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
  at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
  at java.base/java.io.FileInputStream.<init>(FileInputStream.java:112)
  at java.base/sun.net.www.protocol.file.FileURLConnection.connect(FileURLConnection.java:86)
  at java.base/sun.net.www.protocol.file.FileURLConnection.getInputStream(FileURLConnection.java:184)
  at uk.ac.starlink.util.ContentCoding$1.getInputStream(ContentCoding.java:151)
  at uk.ac.starlink.util.URLDataSource.getRawInputStream(URLDataSource.java:98)
  at uk.ac.starlink.util.DataSource.initialise(DataSource.java:257)
  at uk.ac.starlink.util.DataSource.getIntro(DataSource.java:372)
  at uk.ac.starlink.fits.FitsTableBuilder.makeStarTable(FitsTableBuilder.java:109)
  at uk.ac.starlink.votable.VOTableDOMBuilder$Worker.processFitsHref(VOTableDOMBuilder.java:146)
```
To me, this looks a bit mystic. Relevant Debian bug is [Debian#912346](https://bugs.debian.org/912346).

**Comment by mbtaylor** [comment]
Silly bug, easy fix.  Addressed at 5a09c37c82be.  I haven't actually tested this on JDK11, but I'm reasonably confident it should sort it out - please check and close if so.  Thanks for the report.

**Comment by olebole** [comment]
Yes, this works fine for me. Thank you for the quick fix!
